arduino/libraries/librariesmanager/install.go

```diff
@@ -239,3 +251,23 @@ func parseGitURL(gitURL string) (string, error) {
 	}
 	return res, nil
 }
+
+// validateLibrary verifies the dir contains a valid library, meaning it has both
+// an header <name>.h, either in src or root folder, and a library.properties file
+func validateLibrary(name string, dir *paths.Path) error {
+	// Verify library contains library header.
+	// Checks also root folder for legacy reasons.
+	// For more info see:
+	// https://arduino.github.io/arduino-cli/latest/library-specification/#source-code
+	libraryHeader := name + ".h"
+	if !dir.Join("src", libraryHeader).Exist() && !dir.Join(libraryHeader).Exist() {
+		return fmt.Errorf(`library is not valid: missing header file "%s"`, libraryHeader)
+	}
+
+	// Verifies library contains library.properties
+	if !dir.Join("library.properties").Exist() {
```

Comment:
maybe would be the case to check also if the `library.property` is not empty? Or includes the required fields?

Reply:
Not really sure about that, ideally that's Arduino Lint job.
Bringing the lint tool as a dependency to check on these things would be a good idea but the lint tool has the CLI as a dependency, it would open a huge can of worms I think.

Reply:
Nevermind, you are right, probably was a bad idea :space_invader: